pkg/vmprovider/providers/vsphere2/session/session_vm_update.go

```diff
@@ -777,32 +782,206 @@ func (s *Session) attachClusterModule(
 	return s.Client.ClusterModuleClient().AddMoRefToModule(vmCtx, moduleUUID, resVM.MoRef())
 }
 
+func (s *Session) poweredOffVirtualMachine(
```

Comment:
This is a very confusing function name. Can you please rename it to be something more descriptive? Or add a comment?